doc/content/devices/mac-settings/_index.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "MAC Settings"
+description: ""
+---
+
+This section provides guidelines for configuring MAC settings for end devices from the CLI.
+
+<!--more-->
+
+{{< cli-only >}}
+
+### MAC Settings and MAC State
+
+`mac-settings` on {{% tts %}} are configurable per end device. To configure persistent MAC settings, make changes to `mac-settings.desired_<parameter>`. Updates to `mac-settings.desired_<parameter>` take effect on device creation, after OTAA join or ABP FCnt reset, ResetInd, or after MAC state reset.
+
+`mac-state` can be used to test MAC settings immediately. To update settings during the current session, make changes to the `mac-state.desired_parameters.<parameter>`. Updates to the `mac-state.desired_parameters.<parameter>` are lost on reset.
+
+The expected procedure for testing and updating settings is:
+
+1. Modify `mac-state.desired_parameters.<parameter>` to see changes in the current session
+2. Test that everything works as expected
+3. Modify `mac-settings.desired_<parameter>` to make the change permanent
+
+If no settings are provided on device creation or unset, defaults are taken from [Network Server Configuration]({{< ref src="/reference/configuration/network-server" >}}).
+
+### Available MAC settings
+
+Run the following command to get a list of all available MAC settings and available parameter values:
+
+```bash
+$ ttn-lw-cli end-devices update --help
+```
+
+You can also refer to the [End Device API Reference page]({{< ref "reference/api/end_device#message:MACSettings" >}}) for documentation on the available MAC settings and MAC state parameters.
+
+### Class Specific Settings
+
+Settings that are useful based on device class are:
+
+All devices:
+- `mac-settings.factory-preset-frequencies`
```

Comment:
Should we write a short description of what the setting does? Also below?

Ideally imported/copied from the (generated) proto descriptions.

Would be nice if @rvolosatovs could improve documentation for these in the `.proto` files, because now this one just says "List of factory-preset frequencies.". From what I understand this gives the frequencies on which the Network Server accepts Uplink messages from ABP devices or Join Requests from OTAA devices.

Reply:
@rvolosatovs can you give me a quick example configuration I can add, and either expand the `.proto` descriptions or paste them here so I can?

Reply:
Let's make it part of https://github.com/TheThingsNetwork/lorawan-stack/issues/2269

A quick example could be an ABP device using TTN FP frequencies as `factory-preset-frequencies`

Reply:
is that something like `ttn-lw-cli devices update <device_id> --mac-settings.factory-preset-frequencies 868100000 868300000 868500000 867100000 867300000 867500000 867700000 867900000 868800000`

Reply:
I think you have 1 too much (868800000) - it should be 8 frequencies
Also, I don't think the frequencies should be space-delimited, but I may be wrong, @htdvisser ?
Also note that for ABP this should be specified on `create`, otherwise these will only be effective after reset

Reply:
For ABP devices that use TTN's Arduino SDK it's indeed the first 8 from that list (so not 868.8).

From the [documentation of pflag](https://pkg.go.dev/github.com/spf13/pflag?tab=doc#StringSlice) I suspect it parses a comma-separated list, which also makes sense, because CLI arguments split on space.